cmake/CMakeLists.txt

```diff
@@ -538,6 +538,13 @@ if(PKG_USER-NETCDF)
 endif()
 
 if(PKG_USER-SMD)
+  # hack alert!
+  # something defines the Eigen3_DIR variable, but it has no effect
```

Comment:
I won't say it has _no effect_, it just one more variable to help CMake find things.

If you point `Eigen3_DIR` to where `Eigen3Config.cmake` and `Eigen3ConfigVersion.cmake` are, CMake will use the `EIGEN3_INCLUDE_DIR` value given in `Eigen3Config.cmake`.